hand.py

```diff
@@ -33,9 +37,211 @@ def __repr__(self):
             self.hand_list
         )
 
+
 def generate_hand(deck):
     """adds a card to a hand
     >>> generate_hand(Deck([Card('H', '7'), Card('H', '8')]))
     Hand([Card('H', '8'), Card('H', '7')])
     """
-    return Hand([draw_card_from_deck(deck), draw_card_from_deck(deck)])
\ No newline at end of file
+    return Hand([draw_card_from_deck(deck), draw_card_from_deck(deck)])
+
+# def compare_hands(hand_1, hand_2):
+#     """compares hands to check for a winner"""
+#     ##win conditions royal flush > straight flush > flush > straight >
+#     Royal flush.A, K, Q, J, 10, all
+#     Straight flush
+#     Four of akind
+#     Full house
+#     Flush
+#     Straight
+#     Three of a kind
+#     pair
+
+
+def _check_for_4_of_a_kind(hand):
+    """checks to see if the hand has four like cards
+
+    >>> _check_for_4_of_a_kind(Hand([Card('H', '10'), Card('C', '10'), Card('S', '10'), Card('D', '10'), Card('H', 'A')]))
+    True
+    >>> _check_for_4_of_a_kind(Hand([Card('H', '9'), Card('C', '10'), Card('S', '10'), Card('D', '10'), Card('H', 'A')]))
+    False
+    """
+    output = False
+    grouping = group_by(hand.hand_list, rank_key)
+    for item in grouping:
+        grouping[item] = len(grouping[item])
+    for item in grouping:
+        if grouping[item] == 4:
+            output = True
```

Comment:
Almost all of the for-loops in this file could be re-written as calls to `any()` and that matches the semantics of what you're doing, searching.